File: messaging-service/src/main/java/com/yazikochesalna/messagingservice/config/WebClientConfig.java
package com.yazikochesalna.messagingservice.config;

import com.yazikochesalna.messagingservice.config.properties.ChatServiceProperties;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.client.reactive.ReactorClientHttpConnector;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.netty.http.client.HttpClient;

@Configuration
@RequiredArgsConstructor
public class WebClientConfig {

    private final ChatServiceProperties chatServiceProperties;

    @Bean
    public WebClient chatServiceWebClient() {
        return WebClient.builder()
                .baseUrl(chatServiceProperties.getUrl())
                .clientConnector(new ReactorClientHttpConnector(HttpClient.create())).build();
    }
}

File: message-storage-service/src/main/kotlin/com/yazikochesalna/messagestorageservice/config/WebClientConfig.kt
package com.yazikochesalna.messagestorageservice.config

import com.yazikochesalna.messagestorageservice.config.properties.ChatServiceProperties
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.function.client.WebClient

@Configuration
open class WebClientConfig(
) {
    @Bean
    open fun webClient(): WebClient {
        return WebClient.builder()
            //.defaultHeaders { headers -> headers.setBearerAuth("") }
            .build()
    }
}